tidy3d/components/medium.py

```diff
@@ -416,23 +531,79 @@ class CustomMedium(AbstractMedium):
     >>> eps = dielectric.eps_model(200e12)
     """
 
+    # permittivity: Optional[SpatialDataArray] = pd.Field(
```

Comment:
Do we want to either:
1. Handle permittivity and conductivity  in `CustomMedium`
    a. Uncomment these fields
    b. Validate that if `eps_dataset` is provided, it gets converted to `PermittivityConductivity`. Warn user to use an anisotropic custom medium if they want this functionality. (breaks backwards compatibility?)
    c. Use `permittivity` and `conductivity` internally instead of `eps_dataset`.
    c. Provide eps_dataset as a `@property` generated from `permittivity` and `conductivity` to help with backwards compatibility.

or 
2. Provide a new `CustomIsotropicMedium` with `permittivity` and `conductivity` fields and just warn user of current `CustomMedium` that it will be renamed to `CustomAnisotropicMedium` in the future and to use `CustomIsotropicMedium` for now if that's what they want to use this class for?

Reply:
Based on our previous discussion, the plan is to use 2. The commented part here will be replaced by that.

Reply:
ok and what was the plan to handle if the user has some legacy code using eps_dataset with diagonally anisotropic custom medium? Are we planning to just not support that until CustomAnisotropicMedium is introduced? 

in that case, we'd have to error right away and tell them that custom anisotropic medium is no longer supported?

Another option is to instead have `eps_dataset` be the internally used field, convert supplied `permittivity` and `conductivity` to an `eps_dataset` and also provide `permittivity` and `conductivity` as properties. Maybe this is the only way to avoid breaking backwards compatibility

Reply:
Now I add a class `CustomIsotropicMedium` which is only internal in v2.0, and will be renamed as `CustomMedium` in v3.0. In v2.0, `CustomMedium` has a cached property `_medium` which will be a `CustomIsotropicMedium` if `permittivity` and `conductivity` are provided, and `CustomAnisotropicMedium` if `eps_dataset` is supplied. What do you think of the current structure?

One immediate improvement is to have `_medium` as CustomIstropicMedium if the components are identical in `eps_dataset`. That's an easy update that I'll do soon.

Reply:
I think it's up to you since you understand the internals a lot better. As long as the user will be able to make a `CustomMedium(eps_dataset=...)` and still access the `.eps_dataset` and have the internals work as before, I think it's fine either way. We could also just have `CustomIsotropicMedium` maintain an `eps_dataset` (or property for `CustomMedium`) and use the internals from the current `CustomMedium` class.